apps/src/JSInterpreter.js

```diff
@@ -530,7 +530,7 @@ JSInterpreter.prototype.executeInterpreter = function (firstStep, runUntilCallba
     }
     this.executionError = safeStepInterpreter(this);
     if (!this.executionError && this.interpreter.stateStack.length) {
-      var state = this.interpreter.stateStack[0], nodeType = state.node.type;
+      var state = this.interpreter.stateStack[this.interpreter.stateStack.length - 1], nodeType = state.node.type;
```

Comment:
Whoops - I don't think we're enforcing this with the linter yet, but prefer one initialized declaration per statement/line.  I'd also make these `const` if we can, especially since one is derived from the other.